sigstore-java/src/test/java/dev/sigstore/tuf/TufClientTest.java

```diff
@@ -0,0 +1,182 @@
+    }
+  }
+
+  @Test
+  public void testRootUpdate_expiredRoot()
+      throws IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException {
+    setupMirror("remote-repo-expired", "2.root.json");
+    var client = createTufClient();
+    try {
+      client.updateRoot(tufTestData.resolve("trusted-root.json"), new URL(remoteUrl), localStore);
+      fail();
+    } catch (RootExpiredException e) {
+      assertEquals(ZonedDateTime.parse(TEST_STATIC_UPDATE_TIME), e.getUpdateTime());
+      // straight from remote-repo-expired/2.root.json
+      assertEquals(
+          ZonedDateTime.parse("2022-05-11T19:09:02.663975009Z"), e.getRootExpirationTime());
+    }
+  }
+
+  @Test
+  public void testRootUpdate_inconsistentVersion()
+      throws IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException {
+    setupMirror("remote-repo-inconsistent-version", "2.root.json");
+    var client = createTufClient();
+    try {
+      client.updateRoot(tufTestData.resolve("trusted-root.json"), new URL(remoteUrl), localStore);
+      fail();
+    } catch (RoleVersionException e) {
+      assertEquals(2, e.getExpectedVersion());
+      assertEquals(3, e.getFoundVersion());
+    }
+  }
+
+  @Test
+  public void testRootUpdate_metaFileTooBig()
+      throws IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException {
+    setupMirror("remote-repo-meta-file-too-big", "2.root.json");
+    var client = createTufClient();
+    try {
+      client.updateRoot(tufTestData.resolve("trusted-root.json"), new URL(remoteUrl), localStore);
+      fail();
+    } catch (MetaFileExceedsMaxException e) {
+    }
+  }
+
+  @NotNull
+  private static TufClient createTufClient() {
+    TufClient client = new TufClient();
+    // set a fixed time to ensure test results are reproducible.
+    client.clock = Clock.fixed(Instant.parse(TEST_STATIC_UPDATE_TIME), ZoneOffset.UTC);
+    return client;
+  }
+
+  private void setupMirror(String repoFolder, String... files) throws IOException {
+    for (String file : files) {
+      Files.copy(tufTestData.resolve(repoFolder).resolve(file), localMirror.resolve(file));
+    }
+  }
+
+  private void assertRootNotExpired(Root root) {
+    assertTrue(
+        root.getSignedMeta()
+            .getExpiresAsDate()
+            .isAfter(ZonedDateTime.parse(TEST_STATIC_UPDATE_TIME)));
+  }
+
+  private void assertRootVersionIncreased(Root oldRoot, Root newRoot) throws IOException {
+    assertTrue(oldRoot.getSignedMeta().getVersion() <= newRoot.getSignedMeta().getVersion());
+  }
+
+  private void assertStoreContains(String resource) {
+    assertTrue(localStore.resolve(resource).toFile().exists());
+  }
+
+  Root loadRoot(Path rootPath) throws IOException {
+    return GsonSupplier.GSON.get().fromJson(Files.readString(rootPath), Root.class);
+  }
+
+  @AfterEach
+  void clearLocalMirror() {
+    for (File file : localMirror.toFile().listFiles()) {
```

Comment:
I thought `@TempDir` did cleanup itself?

Reply:
I think it's test suite scoped?  Doesn't seem to clean itself between tests.

Reply:
Oh I see it's static and you only want to start the remote resource server once?

Reply:
yeah exactly.